src/_zkapauthorizer/model.py

```diff
@@ -778,6 +779,32 @@ def get_latest_lease_maintenance_activity(self, cursor):
             parse_datetime(finished, delimiter=" "),
         )
 
+    @with_cursor
+    def add_event(self, cursor, sql_statement: str):
+        """
+        Add a new change to the event-log.
+        """
+        cursor.execute(
+            """
+            INSERT INTO [event-stream]([statement]) VALUES (?)
+            """,
+            (sql_statement,),
+        )
+
+    @with_cursor
+    def get_events(self, cursor):
+        """
+        Return all events currently in our event-log.
+        """
+        rows = cursor.execute(
+            """
+            SELECT [sequence-number], [statement]
+            FROM [event-stream]
+            """
+        ).fetchall()
+
+        return EventStream(changes=tuple(Change(seq, stmt) for seq, stmt in rows))
```

Comment:
Hmm.  Feels weird but ... an immutable container that preserves order... yea, I guess that's a tuple.